packages/whenable/README.md

```diff
@@ -0,0 +1,69 @@
+# Whenables
+
+Native promises are not compatible with Agoric's durable stores, which means that on the Agoric platform, such promises disconnect their clients when their creator vat is upgraded.  Whenables are objects that represent promises that can be stored durably, this package also provides a `when` operator to allow clients to tolerate upgrades of whenable-hosting vats, as well as a `watch` operator to subscribe to a promise or whenable in a way that survives upgrades of both the creator and subscribing client vats.
+
+## Whenable Consumer
+
+If your vat is a consumer of promises that are unexpectedly fulfilling to a Whenable (something like):
+
+```js
+import { E } from '@endo/far';
+
+const a = await w;
+const b = await E(w).something(...args);
+console.log('Here they are:', { a, b });
+```
+
+Produces output like:
+```console
+Here they are: {
+  a: Object [Whenable] {
+    payload: { whenableV0: Object [Alleged: WhenableInternalsKit whenableV0] {} }
+  },
+  b: Object [Whenable] {
+    payload: { whenableV0: Object [Alleged: WhenableInternalsKit whenableV0] {} }
+  }
+}
+```
+
+you can change the `await w` into `await when(w)` to convert a chain of whenables to its final settlement, and use the exported `E` to do the same unwrapping implicitly:
+
+```js
+import { E, when } from '@agoric/whenable/heap.js';
+[...]
+const a = await when(w);
+const b = await E(w).something(...args);
+// Produces the expected results.
```

Comment:
would be nice for this to be captured as a test.

Reply:
Done!  PTAL at `agoric-sdk/packages/internal/test/test-whenable.js`.